### ISSUE TITLE
Figures Do Not Work as Documented

### ISSUE THREAD

**Issue by jessebett** [JunoLab/Weave.jl#251]
The following snippets included in a `test.jmd` with the build command. Note I'm using `"""` to indicate code blocks as Github intercepts the markdown.
```julia
weave(papersdir("test.jmd"), 
  out_path=:doc,
  doctype = "md2pdf"
  )
```
demonstrate the various ways Figures are not working.

```julia
"""julia
using Plots

plot(1:10)
"""
```
This works, as expected. The plot takes up 100% of the page width.

1. 
```julia
"""julia
using Plots
"""
![A plot](`j plot(1:10)`)
```

Fails with error 
```
ERROR: MethodError: no method matching write(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Weave.InlineCode)
Closest candidates are:
  write(::IO, ::Any) at io.jl:582
  write(::IO, ::Any, ::Any...) at io.jl:584
  write(::IO, ::Complex) at complex.jl:217
  ...
```
Despite this being [directly from the documentation](http://weavejl.mpastell.com/stable/usage/#Inline-code-1).

2. This is likely the same error as 1, but:
```julia
"""julia
using Plots

p = plot(1:10);
"""
![A plot](`j display(p)`)
```

3. Trying any of the [chunk width options for figures](http://weavejl.mpastell.com/stable/chunk_options/#Options-for-figures-1) and none of them work:
```julia
---
options:
      out_width : 50%
---

"""julia
using Plots

p = plot(1:10);
display(p)
"""
```

Fails to compile with 
```
┌ Warning: Error converting document to pdf. Try running latex manually
```
And this does not work for the other documented option `\\0.5linewidth`.
It does work for `0.5\linewidth`

4. Finally, I have a more complicated plot which looks like the `display(p)` example. However, after running weave with the error
```
┌ Warning: Error converting document to pdf. Try running latex manually
```
I get an empty figure environment with just a printed path-name 
`path/to/weavefile/jl_N8rejs/file_name_22_1.pdf` but there is no folder `jl_N8rejs` and the figure was not correctly weaved into the file.

I have tried to solve this by instead `savefig(p,"fig_path.pdf")` and then in markdown `![My Figure]("fig_path.pdf")` but I am unable to resize that figure as in issue 3. I do not know what is causing this.

TL;DR:
1. Unable to plot inline with documented `![]()` syntax
2. Unable to `display` previously generated plot with `![]()`
3. `out_width` for figures incorrectly documented
4. unknown plotting error

**Comment by jessebett** [comment]
Actually, still trying to understand 4, I realized that 3 is worse than I originally state.

```julia
---
options:
      out_width : 0.5\linewidth
---

"""julia
using Plots

p = plot(1:10);
savefig(p,"testfig.pdf")
"""

![Test Figure](testfig.pdf)
```

This will correctly include the test figure, but the `out_width` option is completely ignored. I see no way to change the width of this figure.
